File: test/fixtures/enums/android/src/main/java/com/example/enums/EnumsPlugin.java
package com.example.enums;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Enums")
public class EnumsPlugin extends Plugin {

    private String currentMode = "native";

    @PluginMethod
    public void configure(PluginCall call) {
        String mode = call.getString("mode");
        currentMode = mode;
        String style = call.getString("style", "DARK");
        JSObject data = new JSObject();
        data.put("mode", mode);
        notifyListeners("modeDetected", data);
        call.resolve();
    }

    @PluginMethod
    public void getMode(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", currentMode);
        call.resolve(ret);
    }

    @PluginMethod
    public void echoMode(PluginCall call) {
        String mode = call.getString("mode");
        String style = call.getString("style", "LIGHT");
        JSObject ret = new JSObject();
        ret.put("mode", mode);
        ret.put("style", style);
        ret.put("origin", "north");
        call.resolve(ret);
    }
}

File: test/fixtures/enums/ios/Sources/EnumsPlugin.swift
import Foundation
import Capacitor

@objc(EnumsPlugin)
public class EnumsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "EnumsPlugin"
    public let jsName = "Enums"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "configure", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getMode", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "echoMode", returnType: CAPPluginReturnPromise)
    ]

    private var currentMode = "native"

    @objc func configure(_ call: CAPPluginCall) {
        let mode = call.getString("mode")
        currentMode = mode
        if let style = call.getString("style") {
            print("plugin2valdi fixture: style", style)
        }
        notifyListeners("modeDetected", data: ["mode": mode])
        call.resolve()
    }

    @objc func getMode(_ call: CAPPluginCall) {
        call.resolve(["value": currentMode])
    }

    @objc func echoMode(_ call: CAPPluginCall) {
        let mode = call.getString("mode")
        var style: String? = nil
        if let s = call.getString("style") { style = s }
        var ret: [String: Any] = ["mode": mode, "origin": "north"]
        if let style = style { ret["style"] = style }
        call.resolve(ret)
    }
}
